pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1049,21 +1164,91 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListenerForPortOrUDS(l
 	var ret bool
 	var opts []*filterChainOpts
 
+	conflictType := NoConflict
+
 	switch pluginParams.ListenerProtocol {
 	case plugin.ListenerProtocolHTTP:
-		if ret, opts = configgen.buildSidecarOutboundHTTPListenerOptsForPortOrUDS(&listenerMapKey, &currentListenerEntry,
+		if ret, opts = configgen.buildSidecarOutboundHTTPListenerOptsForPortOrUDS(node, &listenerMapKey, &currentListenerEntry,
 			&listenerOpts, pluginParams, listenerMap, actualWildcard); !ret {
 			return
 		}
 
+		// Check if conflict happens
+		if util.IsProxyVersionGE13(node) && currentListenerEntry != nil {
+			// Build HTTP listener. If current listener entry is using HTTP or protocol sniffing,
+			// append the service. Otherwise (TCP), change current listener to use protocol sniffing.
+			if currentListenerEntry.protocol.IsHTTP() {
+				conflictType = HTTPOverHTTP
```

Comment:
this can never occur.

Reply:
Why? I think user can configure multiple services on the same port. 

Before in building HTTP filter chain opts, we also check HTTP over HTTP. Now we skip the conflict checking if istio version is >= 1.3. So I think we need to handle conflict here.

Reply:
we check for http over http and then ignore it.. The reason is becuase for HTTP, we send RDS with all the hosts on that port. And HTTP protocol allows us to distinguish between different cases using the Authority header.

Reply:
We append the service and ignore the conflict before. Now we don't append service when we detect HTTP over HTTP immediately. I move the append logic in the following switch case.